src/test/java/org/zeroturnaround/zip/ZipPreserveTimeStampTest.java

```diff
@@ -0,0 +1,81 @@
+package org.zeroturnaround.zip;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.zeroturnaround.zip.commons.FileUtils;
+
+/**
+ *    Copyright (C) 2012 ZeroTurnaround LLC <support@zeroturnaround.com>
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+import junit.framework.TestCase;
+
+/**
+ * These tests shouldn't be run with JDK8 as we preserve
+ * lastModified, create, lastAccessed times for JDK8 and
+ * the tests here that target the older API would fail.
+ */
+@RunWith(SkipForJava8Runner.class)
+public class ZipPreserveTimeStampTest extends TestCase {
+
+  @Test
+  public void testPreservingTimestamps() throws IOException {
```

Comment:
Yeah, I took the existing tests and made them work. Didn't put effort into analysis whether the tests are any good. I'll take a stab at them.